Dockerfile

```diff
@@ -0,0 +1,64 @@
+###############
+# Build stage #
+###############
+
+FROM  elixir:1.5-alpine
+
+WORKDIR /build
+
+# We need hex and rebar build tools installed.
+RUN mix local.hex --force && mix local.rebar --force
+
+# Copy all the source files. We can't just use `COPY . .` because that would
+# also copy deps, _build, and various other things that would cause trouble
+# with the build. Also, `COPY dir .` will copy the contents of the dir instead
+# of the dir itself, so we have to manually copy each dir into its own
+# destination.
+COPY mix.* ./
+COPY config ./config
+COPY lib ./lib
+COPY test ./test
+COPY rel ./rel
+
+# Build the release to put in the next image.
+ENV MIX_ENV=prod
+RUN mix deps.get
+RUN mix compile
+RUN mix release --env=prod
+
+
+###############
+# Image stage #
+###############
+
+# We don't actually need Erlang/Elixir installed, because it's all included in
+# the release package. Thus, we start from the base alpine image.
+FROM alpine:3.7
+
+# We need bash for the generated scripts, tini for signal propagation, and
+# openssl for crypto.
+RUN apk add --no-cache bash tini openssl
+
+WORKDIR /app
+
+# Get the release we built earlier from the build container.
+COPY --from=0 /build/_build/prod/rel/marathon_event_exporter/ ./
+
+RUN ls -lh
+
+# We need runtime write access to /app/var as a non-root user.
+RUN mkdir var && chown nobody var
+
+# Run as non-root.
+USER nobody
```

Comment:
I would kind of prefer a specific user for this rather than `nobody` but actually not sure what the best-practice is.

Reply:
I stuck with `nobody` because it already exists. Given that the container only runs one thing, does the user really matter that much (as long as it's not root)?